### PR TITLE
Hotfix dock.lost

### PR DESCRIPTION
**Pull request by anandkumarpatel**
- make input to dockerode uniform
- updated some modules to ES6 to fix something that turned out to be cyclic dependencies

### CODE REVIEW

**Comment by anandkumarpatel** [comment]
snoop-a-loop$ npm start -- --gamma

> snoop-a-loop@1.0.0 start /Users/anandkumarpatel/run/snoop-a-loop
> ./node_modules/mocha/bin/mocha -t 10000 -b index.js "--gamma"
- should connect to the isolated container from the isolated branch
  Cleanup
    ✓ should fetch the instances (905ms)
    ✓ should delete/destroy the non-repo container (203ms)
    ✓ should delete/destroy the repo container (263ms)
1. New Service Containers
   Creating Container
     ✓ should fetch all template containers (902ms)
     ✓ should copy the source instance (1799ms)
     ✓ should create the build (202ms)
     ✓ should build the build (714ms)
     ✓ should create an instance (309ms)
   Working Container
     ✓ should have a dockerContainer (2520ms)
     ✓ should get logs for that container (90ms)
     ✓ should be succsefully built (1010ms)
     ✓ should have a working terminal (1743ms)
2. New Repository Containers
   Create A Container
     Github
       ✓ should create a github org
       ✓ should fetch a github branch (500ms)
       ✓ should fetch a github repo branch (551ms)
     Source Context
       ✓ should fetch the source context (201ms)
       ✓ should fetch the source context versions (193ms)
     Context & Context Versions
       ✓ should create a context (204ms)
       ✓ should create a context version (918ms)
       ✓ should copy the files (3393ms)
       ✓ should create an AppCodeVersion (1592ms)
     Builds & Instances
       ✓ should create a build for a context version (420ms)
       ✓ should build the build (385ms)
       ✓ should create an instance (506ms)
   Working Container
     ✓ should have a dockerContainer (3031ms)
     ✓ should get logs for that container (115ms)
     ✓ should be successfully built (1010ms)
     ✓ should have a working terminal (1243ms)
3. New Repository Containers created using a mirrored docker file
   Create A Container
     Github
       ✓ should create a github org
       ✓ should fetch a github branch (542ms)
       ✓ should fetch a github repo branch (541ms)
     Source Context
       ✓ should fetch the source context (177ms)
       ✓ should fetch the source context versions (189ms)
     Context & Context Versions
       ✓ should create a context (182ms)
       ✓ should create a context version (813ms)
       ✓ should update the context version when the dockerfile is mirrored (392ms)
       ✓ should fetch the stack analysis (1143ms)
       ✓ should copy the files (2715ms)
       ✓ should create an AppCodeVersion (1147ms)
     Builds & Instances
       ✓ should create a build for a context version (381ms)
       ✓ should build the build (261ms)
       ✓ should create an instance (555ms)
   Working Container
     ✓ should have a dockerContainer (3527ms)
     ✓ should get logs for that container (149ms)
     ✓ should be successfully built (1003ms)
     ✓ should have a working terminal (1161ms)
     ✓ should reflect the mirrored dockerfile configuration (1626ms)
4. Rebuild Repo Container
   Rebuilding without Cache
     ✓ should deep copy the build (480ms)
     ✓ should rebuild the instance without cache (723ms)
   Working Container
     ✓ should have a container (7050ms)
     ✓ should get logs for that container (272ms)
     ✓ should be succsefully built (1009ms)
     ✓ should have a working terminal (250ms)
5. Github Webhooks
   Creating Branch Container
     ✓ should created a new branch (929ms)
     ✓ should create a new instance with the branch name (5609ms)
   Working Container
     ✓ should have a container (501ms)
     ✓ should get logs for that container (278ms)
     ✓ should be succsefully built (1012ms)
     ✓ should have a working terminal (1268ms)
6. Isolation
   Create Container To Isolate
     Create A Container
       Github
         - should create a github org
         - should fetch a github branch
         - should fetch a github repo branch
       Source Context
         - should fetch the source context
         - should fetch the source context versions
       Context & Context Versions
         - should create a context
         - should create a context version
         - should fetch the stack analysis
         - should copy the files
         - should create an AppCodeVersion
       Builds & Instances
         - should create a build for a context version
         - should build the build
         - should create an instance
     Working Container
       - should have a dockerContainer
       - should get logs for that container
       - should be successfully built
       - should have a working terminal
   Create Isolation
   - should create the isolation
   - should create the instances for that isolation
     Isolated Service Container
   
   ```
   - should have a dockerContainer
   - should get logs for that container
   - should be successfully built
   - should have a working terminal
   ```
   
     Isolated Repo Container
       - should have a dockerContainer
       - should get logs for that container
       - should be successfully built
       - should have a working terminal
7. Container To Container DNS
   Repo Instance
     ✓ should connect to the container from the master branch (623ms)
     ✓ should connect to the container from the newly created branch (if not isolated) (296ms)
   Service Instance
   - should connect to the master branch repo instance
   - should connect to the creaated branch repo instance
8. Navi URLs
   Repo Instance
     ✓ should access the main container (268ms)
     ✓ should access the branch container (274ms)
   Service Instance
     1) should connect to the service instance

File: lib/docker-utils.js [hunk 1]
```diff
@@ -15,84 +14,88 @@ const Swarm = require('./swarm')
...
-  .catch((err) => {
-    log.error({ err: err }, 'testEvent - ' + err.message)
-  })
-}
-
-module.exports.toDockerHost = (url) => {
-  return url.replace('http://', '')
-}
+module.exports = class Utils {
+  static testEvent (dockerClient) {
```
Comment (thejsj) [inline lib/docker-utils.js:18]: Nit pick: JSDoc that wasn't here :)

File: lib/docker-utils.js [hunk 2]
```diff
@@ -15,84 +14,88 @@ const Swarm = require('./swarm')
...
+    return swarm.swarmHostExistsAsync(dockerHost)
+      .catch((hostErr) => {
+        log.trace({ err: hostErr }, '_handleInspectError - swarmHostExists error')
+        // if above errors, throw original error
+        throw err
+      })
+      .then((exists) => {
+        if (!exists) {
+          // TODO: cyclic dependency
+          const rabbitmq = require('./rabbitmq')
```
Comment (thejsj) [inline lib/docker-utils.js:86]: Why did you move the `rabbit` here? I do agree with Mr. CC in that this should be at the top of the file.

Comment (thejsj) [inline lib/docker-utils.js:86]: Talked about this IRL. NM.

**Comment by thejsj** [comment]
LGTM
